src/Symfony/Component/DependencyInjection/Compiler/RandomizePrivateServiceIdentifiers.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Alias;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Randomizes private service identifiers, effectively making them unaccessable from outside.
+ *
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+class RandomizePrivateServiceIdentifiers implements CompilerPassInterface
```

Comment:
We disagree :) i think we have compiler passes for that (optional) or otherwise `Container::compile` (by design).

A dumper (eg. `PhpDumper`) dumps the given container as is. Like others do, YAML, XML etc. I dont know why `PhpDumer` would randomize your services, whereas `XmlDumper` doesnt.

Currently it only deals with privates to _hack_ the dumped container (ie. output), saying; here's a list of privates.